academy/ibc-dev/ibcapp_steps.md

```diff
@@ -0,0 +1,425 @@
+
+## Scaffold a Leaderboard chain
+
+By now, you are familiar with scaffolding a chain with Ignite CLI (if not, check out the [Create Your Own Chain](insert_link) section).
+
+Let's scaffold a `leaderboard` chain:
+
+```bash
+ignite scaffold chain github.com/cosmonaut/leaderboard
+```
+
+This creates a chain with `x/leaderboard` a regular SDK module.
+
+Next, scaffold another chain (for example in another git branch) but decide to add the `--no-module` flag:
+
+```bash
+ignite scaffold chain github.com/cosmonaut/leaderboard --no-module
+```
+
+Now we add the `x/leaderboard` module as an IBC module with the `--ibc` flag:
+
+```bash
+ignite scaffold module leaderboard --ibc
+```
+
+The output you see on the terminal when the module has finished scaffolding already gives a sense of what has to be implemented to create an IBC module:
+
+```bash
+modify app/app.go
+modify proto/leaderboard/genesis.proto
+create proto/leaderboard/packet.proto
+modify testutil/keeper/leaderboard.go
+modify x/leaderboard/genesis.go
+create x/leaderboard/module_ibc.go
+create x/leaderboard/types/events_ibc.go
+modify x/leaderboard/types/genesis.go
+modify x/leaderboard/types/keys.go
+```
+
+To have a more detailed view, we can now compare both versions with a git diff.
+
+## IBC application module requirements
+
+What does Ignite CLI do behind the scenes when creating an IBC module for us? What do we need to implement if we want to upgrade a regular custom application module to an IBC enabled module?
+
+The required steps to implement can be found in the [IBC go docs](https://ibc.cosmos.network/main/ibc/apps/apps.html). We find there:
+
+<Highlightbox type="info">
+
+**To have your module interact over IBC you must:**
+
+- implement the `IBCModule` interface, i.e.:
+  - channel (opening) handshake callbacks
+  - channel closing handshake callbacks
+  - packet callbacks
+- bind to a port(s)
+- add keeper methods
+- define your own packet data and acknowledgement structs as well as how to encode/decode them
+- add a route to the IBC router
+
+</Highlightbox>
+
+Now let's take a look at the *git diff* and see if we are able to recognize the steps listed above.
+
+### Implementing the `IBCModule` interface
+
+**NOTE:** For a full explanation, please visit the [IBC go docs](https://ibc.cosmos.network/main/ibc/apps/ibcmodule.html).
+
+The Cosmos SDK expects all IBC modules to implement the [`IBCModule`
+interface](https://github.com/cosmos/ibc-go/tree/main/modules/core/05-port/types/module.go). This interface contains all of the callbacks IBC expects modules to implement. This includes callbacks related to: 
+- channel handshake (`OnChanOpenInit`, `OnChanOpenTry`, `OncChanOpenAck`, `OnChanOpenConfirm`)
+- channel closing (`OnChanCloseInit`, `OnChanCloseConfirm`)
+- packets (`OnRecvPacket`, `OnAcknowledgementPacket` and `OnTimeoutPacket`).
+
+Ignite CLI implements this in the file `x/leaderboard/module_ibc.go`.
+
+//TODO: add collapsible window??
```

Comment:
I might add the entire IBCmodule interface with an <ExpansionPanel> element, because otherwise it's just way too long